commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MeanTest.java

```diff
@@ -0,0 +1,316 @@
+        UniformRandomProvider rng = TestHelper.createRNG();
+        for (int i = 1; i <= 10; i++) {
+            testMeanNonFinite(TestHelper.shuffle(rng, values), expected);
+            testParallelStreamNonFinite(TestHelper.shuffle(rng, values), expected);
+        }
+    }
+
+    static Stream<Arguments> testMeanNonFinite() {
+        return Stream.of(
+            Arguments.of(new double[] {}, Double.NaN),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY},
+                Double.NaN),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.MAX_VALUE},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, -Double.MIN_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NaN, 34.56, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, Double.NaN, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, 89.74, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, 3.14, Double.NaN, Double.NaN},
+                Double.NaN),
+            Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY,
+                Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testCombine")
+    void testCombine(double[] array1, double[] array2) {
+        final double[] combinedArray = TestHelper.concatenate(array1, array2);
+        final double expected = computeExpected(combinedArray);
+        Mean mean1 = Mean.create();
+        Mean mean2 = Mean.create();
+        Arrays.stream(array1).forEach(mean1);
+        Arrays.stream(array2).forEach(mean2);
+        final double mean2BeforeCombine = mean2.getAsDouble();
+        mean1.combine(mean2);
+        TestHelper.assertEquals(expected, mean1.getAsDouble(), 5, () -> "combine");
+        Assertions.assertEquals(mean2BeforeCombine, mean2.getAsDouble());
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testCombine")
+    void testCombineRandomOrder(double[] array1, double[] array2) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        double[] data = TestHelper.concatenate(array1, array2);
```

Comment:
Before you shuffle in one large array, you should perform the loop on the two arrays, shuffling each one.